yt_dlp/extractor/opencast.py

```diff
@@ -0,0 +1,251 @@
+
+        result_obj = {'formats': formats}
+
+        if video_id is not None:
+            result_obj.update({'id': video_id})
+
+        title = video.get('title')
+        if title is not None:
+            result_obj.update({'title': title})
+
+        series = video.get('seriestitle')
+        if series is not None:
+            result_obj.update({'series': series})
+
+        season_id = video.get('series')
+        if season_id is not None:
+            result_obj.update({'season_id': season_id})
+
+        creator = video.get('creators', {}).get('creator')
+        if creator is not None:
+            result_obj.update({'creator': creator})
+
+        timestamp = parse_iso8601(video.get('start'))
+        if timestamp is not None:
+            result_obj.update({'timestamp': timestamp})
+
+        attachments = video.get('attachments', {}).get('attachment', [])
+        if len(attachments) > 0:
+            thumbnail = attachments[0].get('url')
+            result_obj.update({'thumbnail': thumbnail})
+
+        return result_obj
+
+
+class OpencastIE(OpencastBaseIE):
+    _VALID_URL = r'''(?x)
+                    https?://(?P<host>%s)/paella/ui/watch.html\?.*?
+                    id=(?P<id>%s)
+                    ''' % (
+        OpencastBaseIE._INSTANCES_RE,
+        OpencastBaseIE._UUID_RE,
+    )
+
+    _API_BASE = 'https://%s/search/episode.json?id=%s'
+
+    _TEST = {
+        'url': 'https://oc-video1.ruhr-uni-bochum.de/paella/ui/watch.html?id=ed063cd5-72c8-46b5-a60a-569243edcea8',
+        'md5': '554c8e99a90f7be7e874619fcf2a3bc9',
+        'info_dict': {
+            'id': 'ed063cd5-72c8-46b5-a60a-569243edcea8',
+            'ext': 'mp4',
+            'title': '11 - Kryptographie - 24.11.2015',
+            'thumbnail': r're:^https?://.*\.jpg$',
+            'timestamp': 1606208400,
+            'upload_date': '20201124',
+        },
+    }
+
+    def _real_extract(self, url):
+        mobj = re.match(self._VALID_URL, url)
+        host = mobj.group('host')
+        video_id = mobj.group('id')
+
+        api_json = self._call_api(host, video_id, '', note='Downloading video JSON')
+
+        search_results = api_json.get('search-results', {})
+        if 'result' not in search_results:
+            raise ExtractorError('Video was not found')
+
+        result_dict = search_results.get('result', {})
+        if not isinstance(result_dict, dict):
+            raise ExtractorError('More than one video was unexpectedly returned.')
+
+        video = result_dict.get('mediapackage', {})
+
+        result_obj = self._parse_mediapackage(video)
+        return result_obj
+
+
+class OpencastPlaylistIE(OpencastBaseIE):
+    _VALID_URL = r'''(?x)
+                    https?://(?P<host>%s)/engage/ui/index.html\?.*?
+                    epFrom=(?P<id>%s)
+                    ''' % (
+        OpencastBaseIE._INSTANCES_RE,
+        OpencastBaseIE._UUID_RE,
+    )
+
+    _API_BASE = 'https://%s/search/episode.json?sid=%s'
+
+    _TESTS = [
```

Comment:
formatting for this is sus. Idk why isn't the linter catching this 🤔

Reply:
Thanks a lot for your comments! Would you expected the indentation to be like the changed source in l. 186 to 192?

Reply:
both styles are valid for this